### PR TITLE
Improve the handling of CPU microcode values.

### LINKED ISSUE #683: Improve handling CPU microcode values
When using `--evil-json` we often provide an [annotation](https://searchfox.org/mozilla-central/rev/b1e5f2c7c96be36974262551978d54f457db2cae/toolkit/crashreporter/CrashAnnotations.yaml#228-231) that holds the CPU microcode. This should be used to populate the `system_info.cpu_microcode_version` field if we can't extract it from the minidump. Additionally we should print out this value in hexadecimal because that's the format which is most commonly used for listing versions.

### PR DESCRIPTION
**Pull request by afranchuk**
This reads the CPUMicrocodeVersion field from `--evil-json` and uses that if no other microcode version information is available in the minidump. It also changes output to print the microcode version as a hex string.

Closes #683.

One thing that isn't clear: should the output microcode version be 0-padded or not?

### CODE REVIEW

File: minidump-processor/src/process_state.rs
Comment (gabrielesvelto) [inline minidump-processor/src/process_state.rs:921]: Traditionally we haven't padded this value and other places where it's used don't pad it either (think `/proc/cpuinfo` or Intel's documentation) so I'd leave it unpadded for consistency.

File: minidump-processor/src/processor.rs
Comment (gabrielesvelto) [inline minidump-processor/src/processor.rs:488]: Could you split this out in a helper function for better readability?

**Review by gabrielesvelto — CHANGES_REQUESTED** [review]
This is mostly looking good, just a couple of nits to address and it's good to go

File: minidump-processor/src/processor.rs
Comment (afranchuk) [reply minidump-processor/src/processor.rs:488]: To clarify, do you want the conversion from hex split out, or the entire block which determines the microcode version?

Comment (gabrielesvelto) [reply minidump-processor/src/processor.rs:488]: The entire block, something like `get_microcode_version(linux_cpu_info, evil)`.

Comment (afranchuk) [reply minidump-processor/src/processor.rs]: Given the use of local state and unlikelihood of reuse, would it be more appropriate to add comments rather than adding a helper function if readability is an issue? Otherwise you could make the same statement (as far as adding helper functions) for each of the following values that are extracted from streams (`linux_standard_base`, `threads`, etc).

Otherwise, I think `let cpu_microcode_version = get_microcode_version(...)` is redundant in readability (the `let cpu_microcode_version = ...` already makes the functional behavior of the right side of the `=` fairly clear). However, I respect and defer to your final judgement, since you are the code owner :smile:

Comment (gabrielesvelto) [reply minidump-processor/src/processor.rs]: The name was just a suggestion, you could use something more expressive like `microcode_from_cpuinfo_or_evil_json()` or something else that feels appropriate. This isn't because of reuse but to start shrinking this function. We were a bit in a rush when we wrote this so we've ended up with several-hundred-lines-long functions that could use some structure to make it clearer what's going on to someone who reads them the first time.

Comment (afranchuk) [reply minidump-processor/src/processor.rs]: Sure, that sounds like a worthy cause to me!

Comment (afranchuk) [reply minidump-processor/src/processor.rs]: Do you mind if I refactor the code to shrink the function further? Or would it be preferable to do your requested change here and save that for a separate PR?

Comment (gabrielesvelto) [reply minidump-processor/src/processor.rs]: Make it in a separate PR, it's fine if it's just for refactoring, I'd be very happy about that.

**Review by gabrielesvelto — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
